external/storm-kafka/src/jvm/storm/kafka/KafkaSpout.java

```diff
@@ -43,19 +42,19 @@ public MessageAndRealOffset(Message msg, long offset) {
         }
     }
 
-    static enum EmitState {
+    enum EmitState {
         EMITTED_MORE_LEFT,
         EMITTED_END,
         NO_EMITTED
     }
 
-    public static final Logger LOG = LoggerFactory.getLogger(KafkaSpout.class);
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaSpout.class);
 
     SpoutConfig _spoutConfig;
     SpoutOutputCollector _collector;
     PartitionCoordinator _coordinator;
     DynamicPartitionConnections _connections;
-    ZkState _state;
+    PartitionStateManagerFactory _partitionStateManagerFactory;
```

Comment:
instead of a factory, you can make the developer declare the StateStore:
```
public void createTopology() {
    Spout spout = new KafkaSpout(..., new KafkaStateStore(...));
...
}
```

This should keep the KafkaSpout code much simpler and more explicit, and eliminate the need for a factory.